lib/engine/game/g_1822/step/special_track.rb

```diff
@@ -64,9 +60,13 @@ def process_lay_tile(action)
             ability.use!(upgrade: %i[green brown gray].include?(action.tile.color))
             ability.use! if minor_single_use
 
-            if ability.type == :tile_lay && ability.count <= 0 && ability.closed_when_used_up
-              @log << "#{ability.owner.name} closes"
-              ability.owner.close!
+            if ability.type == :tile_lay
+              if ability.count <= 0 && ability.closed_when_used_up
```

Comment:
ability.size

Reply:
This is not the `count` method which can be replaced by `size` if no block is given, but an attr reader for how many times the ability can be used.

https://github.com/tobymao/18xx/blob/faa7f11d904821f1d53ce2e0ce4ed007fcff5253/lib/engine/ability/base.rb#L24

Reply:
woops